src/Application/Features/Participants/EventHandlers/QueueTransferOnMovement.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Cfo.Cats.Application.Common.Interfaces.Locations;
+using Cfo.Cats.Domain.Entities.Participants;
+using Cfo.Cats.Domain.Events;
+
+namespace Cfo.Cats.Application.Features.Participants.EventHandlers;
+
+public class QueueTransferOnMovement(IUnitOfWork unitOfWork,        
+      ILocationService locationService) : INotificationHandler<ParticipantMovedDomainEvent>
+{
+    public async Task Handle(ParticipantMovedDomainEvent notification, CancellationToken cancellationToken)
+    {
+        var now = DateTime.UtcNow;
+        var createOutGoingQueue = false;
+
+        //Get all items off Incoming Queue for the participant
+        var allIncomingQueueEntriesForParticipant = await unitOfWork.DbContext.ParticipantIncomingTransferQueue
+            .Where(q => q.ParticipantId == notification.Item.Id)
+            .ToListAsync(cancellationToken);
+
+        //Are there any items on queue ?
+        if (allIncomingQueueEntriesForParticipant.Any())
+        {
+            //Are there already open items on queue for participant ?
+            var openItem = allIncomingQueueEntriesForParticipant?.Where(q => q.Completed == false).FirstOrDefault();
+            if (openItem != null)
+            {
+                //get list of owners
+                var previouslyOwned = await unitOfWork.DbContext.ParticipantOwnershipHistories
+                    .Where(x => x.ParticipantId == notification.Item.Id
+                    && x.From > DateTime.UtcNow.AddDays(-60))
+                    //.Where(x => x.TenantId.StartsWith(notification.From.Tenants))
+                    .ToListAsync(cancellationToken);               
+
+                //are there previous owners
+                if (previouslyOwned.Any())
+                {
+                    foreach (var previousOwer in previouslyOwned)
+                    {
+                        var ownedLocations = locationService.GetVisibleLocations(previousOwer.TenantId).Select(location => location.Id);
+
+                        //Dont add to Outgoing queue unless particiapnt has been assigned to them in last 3 months. (ownership table)
+                        if (ownedLocations.Any())
+                        {
+                            createOutGoingQueue = true;
+                        }
+                    }
+                }
+
+                //close open item on queue                
+                openItem?.Complete();
+            }
+            //inputqueue item on queue but been closed / assigned
+            else
+            {
+                createOutGoingQueue = true;
+            }
+        }
+        //Effectively first time transferred.
+        else
+        {
+            createOutGoingQueue = true;
+        }
```

Comment:
```suggestion
        var previousIncomingTransfers = await unitOfWork.DbContext.ParticipantIncomingTransferQueue
            .Where(q => q.ParticipantId == notification.Item.Id)
            .ToListAsync(cancellationToken);

        if(previousIncomingTransfers is { Count: 0 })
        {
            createOutGoingQueue = true;
        }
        else
        {
            // Is there an active (in progress) transfer?
            var activeTransfer = previousIncomingTransfers
                .SingleOrDefault(q => q.Completed == false);

            if (activeTransfer is null)
            {
                createOutGoingQueue = true;
            }
            else
            {
                // Complete active transfer
                activeTransfer.Complete();

                // Get list of owners
                var owners = await unitOfWork.DbContext.ParticipantOwnershipHistories
                    .Where(x => x.ParticipantId == notification.Item.Id
                    && x.From > DateTime.UtcNow.AddDays(-60))
                    //.Where(x => x.TenantId.StartsWith(notification.From.Tenants))
                    .ToListAsync(cancellationToken);

                var ownedLocations = owners.SelectMany(o => locationService.GetVisibleLocations(o.TenantId));

                // Dont add to Outgoing queue unless participant has been assigned to them in last 3 months
                if (ownedLocations.Any())
                {
                    createOutGoingQueue = true;
                }

            }
        }
```

I've suggested the above change to make this section somewhat readable. I still don't understand what this is trying to accomplish, however.